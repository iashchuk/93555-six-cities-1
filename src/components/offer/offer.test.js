import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Offer from "./offer.jsx";
import Map from "../map/map.jsx";

const offer = {
  id: 1,
  title: `Beautiful & luxurious apartment at great location`,
  type: `Apartment`,
  price: 120,
  rating: 93,
  isPremium: true,
  isFavorite: false,
  images: [
    `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/16.jpg`
  ],
  host: {
    id: 25,
    name: `Angelina`,
    isPro: true,
    avatarUrl: `img/avatar-angelina.jpg`
  },
  maxAdults: 2,
  bedrooms: 1,
  goods: [
    `Baby seat`,
    `Breakfast`,
    `Air conditioning`,
    `Laptop friendly workspace`,
    `Washer`
  ],
  description: `Relax, rejuvenate and unplug in this ultimate`,
  cityLocation: [52.3909553943508, 4.85309666406198]
};

const offers = [
  {
    id: 1,
    title: `Beautiful & luxurious apartment at great location`,
    image: `apartment-01.jpg`,
    type: `Apartment`,
    price: 120,
    currency: `€`,
    rating: 93,
    isPremium: true,
    location: { latitude: 52.3909553943508, longitude: 4.85309666406198 }
  },
  {
    id: 2,
    title: `Wood and stone place`,
    image: `room.jpg`,
    type: `Private room`,
    price: 80,
    currency: `€`,
    rating: 80,
    isPremium: false,
    location: { latitude: 52.3909553943508, longitude: 4.85309666406198 }
  }
];

const comments = [
  {
    id: 1,
    user: {
      id: 17,
      isPro: false,
      name: `Emely`,
      avatarUrl: `https://es31-server.appspot.com/six-cities/static/avatar/8.jpg`
    },
    rating: 3,
    comment: `We loved it so much, the house, the veiw, the location just great.. Highly reccomend :)`,
    date: `2019-06-03T11:42:26.499Z`
  }
];

const isAuthenticated = true;

const match = {
  params: {
    id: 3
  }
};

describe(`Offer`, () => {
  it(`renders correctly`, () => {
    const getOffer = jest.fn();
    const getComments = jest.fn();
    const setFavoriteAsync = jest.fn();
    const sendReviewAsync = jest.fn();
    Map.prototype.componentDidMount = jest.fn();
    const component = renderer
      .create(
          <Router>
            <Offer
              isAuthenticated={isAuthenticated}
              match={match}
              offer={offer}
              offers={offers}
              comments={comments}
              getOffer={getOffer}
              getComments={getComments}
              setFavoriteAsync={setFavoriteAsync}
              sendReviewAsync={sendReviewAsync}
            />
          </Router>
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
