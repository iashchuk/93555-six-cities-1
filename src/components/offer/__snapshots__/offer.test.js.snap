// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offer renders correctly 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://es31-server.appspot.com/six-cities/static/hotel/2.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://es31-server.appspot.com/six-cities/static/hotel/5.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://es31-server.appspot.com/six-cities/static/hotel/16.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Beautiful & luxurious apartment at great location
          </h1>
          <button
            className="property__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "1860.0000000000002%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            93
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Apartment
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            1
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            2
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            120
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Baby seat
            </li>
            <li
              className="property__inside-item"
            >
              Breakfast
            </li>
            <li
              className="property__inside-item"
            >
              Air conditioning
            </li>
            <li
              className="property__inside-item"
            >
              Laptop friendly workspace
            </li>
            <li
              className="property__inside-item"
            >
              Washer
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="img/avatar-angelina.jpg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
            <span
              className="property__user-status"
            >
              Pro
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              Relax, rejuvenate and unplug in this ultimate
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews ·
            <span
              className="reviews__amount"
            >
              1
            </span>
          </h2>
          <ul
            className="reviews__list"
          >
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="https://es31-server.appspot.com/six-cities/static/avatar/8.jpg"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  Emely
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "60%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  We loved it so much, the house, the veiw, the location just great.. Highly reccomend :)
                </p>
                <time
                  className="reviews__time"
                  dateTime="2019-04-24"
                >
                  June 3rd
                </time>
              </div>
            </li>
          </ul>
        </section>
        <form
          action="#"
          className="reviews__form form"
          method="post"
          onSubmit={[Function]}
        >
          <label
            className="reviews__label form__label"
            htmlFor="review"
          >
            Your review
          </label>
          <div
            className="reviews__rating-form form__rating"
          >
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              disabled={false}
              id="5-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={5}
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="5-stars"
              title="perfect"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              disabled={false}
              id="4-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={4}
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="4-stars"
              title="good"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              disabled={false}
              id="3-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={3}
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="3-stars"
              title="not bad"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              disabled={false}
              id="2-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={2}
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="2-stars"
              title="badly"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              disabled={false}
              id="1-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="1-stars"
              title="terribly"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
          </div>
          <textarea
            className="reviews__textarea form__textarea"
            disabled={false}
            id="review"
            name="review"
            onChange={[Function]}
            placeholder="Tell how was your stay, what you like and what can be improved"
            value=""
          />
          <div
            className="reviews__button-wrapper"
          >
            <p
              className="reviews__help"
            >
              To submit review please make sure to set
              <span
                className="reviews__star"
              >
                rating
              </span>
               and describe your stay with at least 
              <b
                className="reviews__text-amount"
              >
                50 characters
              </b>
              .
            </p>
            <button
              className="reviews__submit form__submit button"
              disabled={true}
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
    <section
      className="property__map map map--offer"
    >
      <div
        className="cities-map"
        id="map"
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="cities__places-list places__list tabs__content"
      >
        <article
          className="cities__place-card place-card"
        >
          <div
            className="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="cities__image-wrapper place-card__image-wrapper"
            onClick={[Function]}
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="200"
              src="apartment-01.jpg"
              width="260"
            />
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  120
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button  button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "93%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/1"
                onClick={[Function]}
              >
                Beautiful & luxurious apartment at great location
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Apartment
            </p>
          </div>
        </article>
        <article
          className="cities__place-card place-card"
        >
          <div
            className="cities__image-wrapper place-card__image-wrapper"
            onClick={[Function]}
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="200"
              src="room.jpg"
              width="260"
            />
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  80
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button  button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "80%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/2"
                onClick={[Function]}
              >
                Wood and stone place
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Private room
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;
